src/SonarScanner.MSBuild.PreProcessor/Roslyn/EmbeddedAnalyzerInstaller.cs

```diff
@@ -83,22 +83,24 @@ public IEnumerable<string> InstallAssemblies(IEnumerable<Plugin> plugins)
             if (!plugins.Any())
             {
                 this.logger.LogInfo(RoslynResources.EAI_NoPluginsSpecified);
-                return Enumerable.Empty<string>(); // nothing to deploy
+                return Enumerable.Empty<AnalyzerPlugin>(); // nothing to deploy
             }
 
             this.logger.LogInfo(RoslynResources.EAI_InstallingAnalyzers);
 
-            var allFiles = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            var analyzerPlugins = new List<AnalyzerPlugin>();
             foreach (var plugin in plugins)
             {
                 var files = GetPluginResourceFiles(plugin);
-                foreach (var file in files)
+                // Don't add the plugin to the list if it doesn't have any assemblies
+                if (files.Any())
```

Comment:
I've added a new unit test for this.

Reply:
Do we really need this check? If the task just adds references to all plugin files then it should work without this check too...